### PR TITLE
library: Add Spinner entry

### PR DESCRIPTION
**Pull request by SoNiC-HeRE**
Added Spinner Demo under User Interface Section in Library
Issue #69

### CODE REVIEW

File: src/app.gresource.xml [hunk 1]
```diff
@@ -37,8 +37,11 @@
     <file>Library/demos/Custom Widget/main.ui</file>
     <file>Library/demos/Custom Widget/main.js</file>
     <file>Library/demos/Custom Widget/main.json</file>
-     <file>Library/demos/Switch/main.json</file>
+    <file>Library/demos/Switch/main.json</file>
     <file>Library/demos/Switch/main.js</file>
     <file>Library/demos/Switch/main.blp</file>
+    <file>Library/demos/Spinner/main.blp</file>
+    <file>Library/demos/Spinner/main.js</file>
+    <file>Library/demos/Spinner/main.json</file>
```
Comment (sonnyp) [inline src/app.gresource.xml:45]: not needed anymore - please merge `main`

https://github.com/sonnyp/Workbench/pull/200

File: src/Library/demos/Spinner/main.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "Spinner",
+  "category": "user_interface",
+  "description": "Display Loading State",
```
Comment (sonnyp) [inline src/Library/demos/Spinner/main.json:4]: only titles should be title cased

```suggestion
  "description": "Display loading state",
```

File: src/Library/demos/Spinner/main.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const start = workbench.builder.get_object("start");
+const stop = workbench.builder.get_object("stop");
+const spinner = workbench.builder.get_object("spinner");
+
+start.connect("clicked", () => {
+  spinner.start();
+  spinner.set_spinning(true);
```
Comment (sonnyp) [inline src/Library/demos/Spinner/main.js:7]: they do the the same thing

also please use properties when possible

```suggestion
  spinner.spinning = false;
```

**Review by sonnyp — CHANGES_REQUESTED** [review]
nice one! :+1: 

Can you merge the stop and play button?

When the spinner spins – it's a pause button
When the spinner does not spin – it's a play button

Ho and center the button please :pray: 


image

File: src/Library/demos/Spinner/main.blp [hunk 4]
```diff
@@ -0,0 +1,49 @@
+using Gtk 4.0;
+using Adw 1;
+
+Adw.StatusPage {
+  title: "Spinner";
+  description: _("Display Loading State");
```
Comment (sonnyp) [inline src/Library/demos/Spinner/main.blp:6]:
```suggestion
  description: _("Display loading state");
```

**Comment by SoNiC-HeRE** [comment]
Sure, I'll make the Changes

**Comment by SoNiC-HeRE** [comment]
Sure, I'll make the necessary changes

File: src/Library/demos/Spinner/main.blp [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    LinkButton{
+      label: "Human Interface Guidelines";
+      uri: "https://developer.gnome.org/hig/patterns/feedback/spinners.html";
+    }
+  }
+}
+
+
```
Comment (sonnyp) [inline src/Library/demos/Spinner/main.blp:46]: fix indentation in this file

File: src/Library/demos/Spinner/main.blp [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+Adw.StatusPage {
+  title: "Spinner";
+  description: _("Display loading state");
+
+  Box {
+    orientation: vertical;
+    halign: center;
+
+    Box {
```
Comment (sonnyp) [inline src/Library/demos/Spinner/main.blp:12]: this box is unecessary

try to keep the amount of UI needed to a minimum

set `halign: center;` on the `spinner` instead

File: src/Library/demos/Spinner/main.blp [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      orientation: vertical;
+      halign: center;
+
+       Spinner spinner {
+          spinning: true;
+          margin-bottom: 30;
+        }
+
+      Button  btn {
+        icon-name: "media-playback-stop";
```
Comment (sonnyp) [inline src/Library/demos/Spinner/main.blp:22]: hmm - I found it difficult to recognize what it is - what do you think of using `media-playback-pause` instead?

**Review by sonnyp — CHANGES_REQUESTED** [review]

File: src/Library/demos/Spinner/main.blp (hunk 7 above)
Comment (SoNiC-HeRE) [reply src/Library/demos/Spinner/main.blp:22]: Yep would look even better I'd update it

File: src/Library/demos/Spinner/main.blp (hunk 6 above)
Comment (sonnyp) [reply src/Library/demos/Spinner/main.blp:12]: sorry - I meant 

set `halign: center;` on the `Button` instead

Comment (SoNiC-HeRE) [reply src/Library/demos/Spinner/main.blp:12]: okay

**Review by sonnyp — CHANGES_REQUESTED** [review]
There is a bug in the program. Let's see if you can find it :)

In addition to reviewing the changes – make sure to test before submitting for review.

**Comment by SoNiC-HeRE** [comment]
> There is a bug in the program. Let's see if you can find it :)
> 
> In addition to reviewing the changes – make sure to test before submitting for review.


I found out an error in main.js i used media-playback-stop instead of media-playback-pause while i used media-playback-pause initially.
Fixed it Now.
Also I used '===' for strict equality to make sure the condition executes only when spinning returns true.

**Review by sonnyp — APPROVED** [review]

**Pull request merged**